File: src/api/src/main/java/fr/iut/uca/model/surveys/Feedback.java
package fr.iut.uca.model.surveys;

import java.util.ArrayList;
import java.time.LocalDate;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

public class Feedback {

    private final String id;

    private String surveyId;

    private final LocalDate createdAt;

    private String author;

    private Question question;

    private final List<String> answers = new ArrayList<>();

    public Feedback(String id, String surveyId,LocalDate createdAt, String author, Question question, List<String> answers) {
        this.id = id;
        this.surveyId = surveyId;
        this.createdAt = createdAt;
        this.author = author;
        this.question = question;
        this.answers.addAll(answers);
    }

    public Feedback(String id, String surveyId,LocalDate createdAt, String author, Question question) {
        this(id, surveyId, createdAt, author, question, Collections.emptyList());
    }

    public String getId() {
        return id;
    }

    public String getSurveyId() {
        return surveyId;
    }

    public void setSurveyId(String surveyId) {
        this.surveyId = surveyId;
    }

    public LocalDate getCreatedAt() {
        return createdAt;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public Question getQuestion() {
        return question;
    }

    public void setQuestion(Question question) {
        this.question = question;
    }

    public List<String> getAnswers() {
        return answers;
    }
}

File: src/app/model/surveys/Feedback.ts
import {Question} from "./Question";

export class Feedback {
  private _id: string
  private _surveyId: string
  private _createdAt: Date
  private _author: string
  private _question: Question
  private _answers: string[]

  constructor(id: string, surveyId: string, createdAt: Date, author: string, question: Question, answers: string[]) {
    this._id = id;
    this._surveyId = surveyId;
    this._createdAt = createdAt;
    this._author = author;
    this._question = question;
    this._answers = answers;
  }

  get id(): string {
    return this._id;
  }

  set id(value: string) {
    this._id = value;
  }

  get surveyId(): string {
    return this._surveyId;
  }

  set surveyId(value: string) {
    this._surveyId = value;
  }

  get createdAt(): Date {
    return this._createdAt;
  }

  set createdAt(value: Date) {
    this._createdAt = value;
  }

  get author(): string {
    return this._author;
  }

  set author(value: string) {
    this._author = value;
  }

  get question(): Question {
    return this._question;
  }

  set question(value: Question) {
    this._question = value;
  }

  get answers(): string[] {
    return this._answers;
  }

  set answers(value: string[]) {
    this._answers = value;
  }
}
